src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/service/ChipsRestService.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.model.ChipsKafkaMessage;
+
+import javax.annotation.PostConstruct;
+import java.util.Collection;
+import java.util.Collections;
+
+@Service
+public class ChipsRestService {
+
+    @Value("${CHIPS_REST_INTERFACES_HOST}")
+    private String chipsRestHost;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    private String chipsRestUrl;
+
+    @PostConstruct
+    void init() {
+        chipsRestUrl = chipsRestHost + "{chips-rest-endpoint}";
+    }
+
+    void sendToChips(Collection<ChipsKafkaMessage> messages) {
+        messages.parallelStream().forEach((message -> {
```

Comment:
I'm wondering if the loop should not be moved out of here - depending on how we will want to implement failed messages, do we want:

a)  to add errored messages to a datat structure that can be picked up after the loop has finished? or
b) add to the error queue as we go through the loop?

In either case we may want to keep error handling code separate from the rest service if possible. What do you think?

Reply:
Where would you move the loop to?

In my mind we would create a custom exception handler and inject it into the restTemplate
So any exception would cause the exception handler to be executed. Which would handle adding the message to the retry topic.
So I guess more b.

Reply:
I was thinking of a service layer that would sit between the consumer class and this rest class and there capture the exceptions and handle the errors, db calls etc. perhaps - this is more about what we do going forward into the next tickets really - but we are doing here had just mad eme think abou that detail.

Reply:
I guess at the moment the Application class is acting like the service, it contains the main loop and receives collections back from incomingMessageConsumer.read(). I think it would be better to then loop through the returned collection and pass each message in to the ChipsRestService, so change ChipsRestService to receive a single message. We could then move the main loop out of Application into a new service that would be the 'heart' of the app. But that would mean Application would simply hand control over to the new service on startup.

Reply:
custom exception handler will work as well, just thinking if ChipsRestService took a single message would it simplify the unit testing?